File: javascript/binary_search/src/binary_search.js
const search = (arr, num) => {
    return search_recursive(arr, 0, arr.length - 1, num);
}

const search_recursive = (arr, start, end, num) => {
    let median = Math.floor((start + end) / 2);
    if ( arr[median] == num) {
        return true;
    } else if (end - start < 1) {
        return false;
    } else if ( arr[median] < num) {
        return search_recursive(arr, median + 1, end, num);
    } else {
        return search_recursive(arr, start, median - 1, num);
    }
}

exports.search = search;

File: python/binary_search/binary_search.py
def search(lst, num):
    return search_recursive(lst, 0, len(lst) - 1, num)

def search_recursive(lst, start, end, num):
    median = (start + end) // 2
    if lst[median] == num:
        return True
    elif end - start < 1:
        return False
    elif lst[median] > num:
        return search_recursive(lst, start, median - 1, num)
    else:
        return search_recursive(lst, median + 1, end, num)
